packages/backend/migrations/20230918113102_rename_payment_pointer_tables.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return Promise.all([
+    knex.schema.renameTable('paymentPointers', 'walletAddresses'),
+    knex.schema.alterTable('walletAddresses', function (table) {
+      table.foreign(['assetId']).references('assets.id')
+      table.unique('url')
+    }),
+    knex.raw(
+      'ALTER INDEX "paymentPointers_pkey" RENAME TO "walletAddresses_pkey"'
+    ),
+    knex.raw(
+      'ALTER INDEX "paymentpointers_url_index" RENAME TO "walletAddresses_url_index"'
+    ),
+    knex.raw(
+      'ALTER INDEX "paymentpointers_processat_index" RENAME TO "walletAddresses_processat_index"'
+    ),
+    knex.raw(
+      'ALTER TABLE "walletAddresses" DROP CONSTRAINT "paymentpointers_url_unique"'
+    ),
+    knex.raw(
+      'ALTER TABLE "walletAddresses" DROP CONSTRAINT "paymentpointers_assetid_foreign"'
+    ),
+    knex.schema.renameTable('paymentPointerKeys', 'walletAddressKeys'),
+    knex.schema.alterTable('walletAddressKeys', function (table) {
+      table.renameColumn('paymentPointerId', 'walletAddressId')
+      table.foreign('walletAddressId').references('walletAddresses.id')
+    }),
+    knex.raw(
+      'ALTER INDEX "paymentPointerKeys_pkey" RENAME TO "walletAddressKeys_pkey"'
+    ),
+    knex.raw(
+      'ALTER TABLE "walletAddressKeys" DROP CONSTRAINT "paymentpointerkeys_paymentpointerid_foreign"'
+    ),
+    knex.schema.alterTable('quotes', function (table) {
+      table.dropForeign(['paymentPointerId'])
+      table.dropIndex(['paymentPointerId', 'createdAt', 'id'])
+      table.renameColumn('paymentPointerId', 'walletAddressId')
+      table.foreign('walletAddressId').references('walletAddresses.id')
+      table.index(['walletAddressId'])
+      table.index(['createdAt'])
+      table.index(['id'])
```

Comment:
The indexes are not aggregated because the original indexes weren't either. Aggregating them produces, for example:
```
Indexes:
"outgoingpayments_walletaddressid_createdat_id_index" btree ("walletAddressId", "createdAt", id)
```

Whereas originally the indexes were not aggregated:
```
"outgoingpayments_paymentpointerid__index" btree ("paymentPointerId")
"outgoingpayments_createdat_index" btree ("createdAt")
"outgoingpayments_id_index" btree (id)
```

So `table.index` is called separately each time to preserve this.

Reply:
> The indexes are not aggregated because the original indexes weren't either.

It seems like they are originally 1 composite index to me but maybe I'm missing something... without running this migration I see `"outgoingpayments_paymentpointerid_createdat_id_index" btree ("paymentPointerId", "createdAt", id)` (same for `quotes`, did not check the rest). If they were seperate indexes I also don't think we'd need to do `table.dropIndex(['paymentPointerId', 'createdAt', 'id'])`, but instead would just drop the payment pointer one (and thus not have to re-add the others).

Reply:
Oh yeah, they're composite. A misreading on my part earlier. So we'll keep the composite dropping & adding of these indexes.